### ISSUE TITLE
Installation problem

### ISSUE THREAD

**Issue by Tirguy** [audreyt/ethercalc#773]
Hello,

I don't really know NodeJS, but I'm currently trying to install EtherCalc on one of my intranet servers. I have enabled the proxy, but I don't have access to everything... However, I've already managed to install others NodeJS based projects and everything works fine.

My server is under Linux CentOS 7 on which I installed more recent versions of some products (git, node, etc...).

My node version is 14.18.2

I cloned the current development version of EtherCalc and put it in /var/www/ethercalc

During the installation, I encountered a first bug: the command that asks to run "make" always reported ""
Makefile indicate : cp node_modules/socialcalc/SocialCalc.js node_modules/socialcalc/dist/SocialCalc.js || true
As "node_modules/socialcalc/SocialCalc.js" does not exist, i've found this file in node_modules/socialcalc/dist/SocialCalc.js and make a copy in node_modules/socialcalc/

A new problem appeared always with the command "make": at the end of the Makefile, node is executed with arguments and notably the argument "--source-map-include-sources", except that this argument doesn't seem to exist (I searched a lot on the Internet and didn't find anything about it): "ERROR: invalid option --source-map-include-sources"

When I remove this option, a new error appears: "ERROR: cannot write source map to STDOUT" which seems to be an error message from the file node_modules/uglify-js/bin/uglifyjs

I don't know how to fix these problems : could someone help me ? or complete the documentation if particular things must be done before the execution of the make command.

Thank you,
Thierry

**Comment by eddyparkinson** [comment]
try:  node app.jsIt looks like it is re-compiling ethercalc rather than running it. Use the command node app.js to run it from the command line.




    On Friday, 31 December 2021, 23:18:22 GMT+10:30, Tirguy ***@***.***> wrote:  
 
 


Hello,

I don't really know NodeJS, but I'm currently trying to install EtherCalc on one of my intranet servers. I have enabled the proxy, but I don't have access to everything... However, I've already managed to install others NodeJS based projects and everything works fine.

My server is under Linux CentOS 7 on which I installed more recent versions of some products (git, node, etc...).

My node version is 14.18.2

I cloned the current development version of EtherCalc and put it in /var/www/ethercalc

During the installation, I encountered a first bug: the command that asks to run "make" always reported ""
Makefile indicate : cp node_modules/socialcalc/SocialCalc.js node_modules/socialcalc/dist/SocialCalc.js || true
As "node_modules/socialcalc/SocialCalc.js" does not exist, i've found this file in node_modules/socialcalc/dist/SocialCalc.js and make a copy in node_modules/socialcalc/

A new problem appeared always with the command "make": at the end of the Makefile, node is executed with arguments and notably the argument "--source-map-include-sources", except that this argument doesn't seem to exist (I searched a lot on the Internet and didn't find anything about it): "ERROR: invalid option --source-map-include-sources"

When I remove this option, a new error appears: "ERROR: cannot write source map to STDOUT" which seems to be an error message from the file node_modules/uglify-js/bin/uglifyjs

I don't know how to fix these problems : could someone help me ? or complete the documentation if particular things must be done before the execution of the make command.

Thank you,
Thierry

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
